src/sentry/migrations/0637_backfill_priority_for_groups.py

```diff
@@ -0,0 +1,125 @@
+from sentry.issues.grouptype import (
+    GroupCategory,
+    PerformanceDurationRegressionGroupType,
+    PerformanceP95EndpointRegressionGroupType,
+    ProfileFunctionRegressionExperimentalType,
+    ProfileFunctionRegressionType,
+    get_group_type_by_type_id,
+)
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.types.group import GroupSubStatus, PriorityLevel
+from sentry.utils.query import RangeQuerySetWrapper
+
+BATCH_SIZE = 100
+
+UPDATE_QUERY = """
+    UPDATE sentry_groupedmessage
+    SET priority = data.priority
+    FROM (VALUES %s) as data (id, priority)
+    WHERE sentry_groupedmessage.id = data.id AND sentry_groupedmessage.priority IS NULL
+"""
+
+logger = logging.getLogger(__name__)
+
+
+def _get_priority_level(group_id, level, type_id, substatus):
+    # TODO: profiling issues
+    group_type = get_group_type_by_type_id(type_id)
+
+    # Replay and Feedback issues are medium priority
+    if group_type.category in [GroupCategory.REPLAY.value, GroupCategory.FEEDBACK.value]:
+        return PriorityLevel.MEDIUM
+
+    # All escalating issues are high priority for all other issue categories
+    if substatus == GroupSubStatus.ESCALATING:
+        return PriorityLevel.HIGH
+
+    if group_type.category == GroupCategory.ERROR.value:
+        if level in [logging.INFO, logging.DEBUG]:
+            return PriorityLevel.LOW
+        elif level == logging.WARNING:
+            return PriorityLevel.MEDIUM
+        elif level in [logging.ERROR, logging.FATAL]:
+            return PriorityLevel.HIGH
+
+        logger.warning('Unknown log level "%s" for group %s', level, group_id)
+        return PriorityLevel.MEDIUM
+
+    if group_type.category == GroupCategory.CRON.value:
+        if level == logging.WARNING:
+            return PriorityLevel.MEDIUM
+
+        return PriorityLevel.HIGH
+
+    if group_type.category == GroupCategory.PERFORMANCE.value:
+        # tbd on the last two
+        if type_id in [
+            PerformanceDurationRegressionGroupType.type_id,
+            PerformanceP95EndpointRegressionGroupType.type_id,
+            ProfileFunctionRegressionExperimentalType.type_id,
+            ProfileFunctionRegressionType.type_id,
+        ]:
+            return PriorityLevel.MEDIUM
+        return PriorityLevel.LOW
+
+    # All other issues are the default medium priority
+    return PriorityLevel.MEDIUM
+
+
+def update_group_priority(apps, schema_editor):
+    Group = apps.get_model("sentry", "Group")
+
+    cursor = connection.cursor()
+    batch = []
+
+    for group_id, data, level, group_type, substatus, priority in RangeQuerySetWrapper(
+        Group.objects.all().values_list("id", "data", "level", "type", "substatus", "priority"),
+        result_value_getter=lambda item: item[0],
+    ):
+        if priority is not None:
+            continue
+
+        priority = _get_priority_level(group_id, level, group_type, substatus)
+        # data.get("metadata", {})["initial_priority"] = priority
```

Comment:
We want to also update the metadata["initial_priority"] field. @wedamija is there a way to do that with the update query? I don't think dicts are supported.

Reply:
That sounds like the right approach to me, just make sure we have tests that load the real model and make sure the metadata is still correct